pkg/render/passthru.go

```diff
@@ -50,3 +60,11 @@ func (p *passthroughComponent) Ready() bool {
 func (p *passthroughComponent) SupportedOSType() rmeta.OSType {
 	return rmeta.OSTypeAny
 }
+
+// NewKeyPairPassthrough is a convenience func for constructing a passthrough to create or clean up secrets in the operator namespace.
+func NewKeyPairPassthrough(keyPair certificatemanagement.KeyPair) Component {
```

Comment:
I'm not sure there's a lot of value to this given that you're already checking UseCertificateManagement() in a bunch of other places and reacting to it.

For now, I'd just remove this function, and later you can follow up with what we were talking about creating a function on the KeyPair object that returns items that need to be created or deleted and hides whether or not this is using cert management.